draganescu/wordpress-develop
# WIP: Try/email address confirm boemedia

## What is it?

This is a PR with a demo of how the "Is your/this admin email still correct" Trac ticket could function. 
Reference Trac Ticket ID: 46349 ([link](https://core.trac.wordpress.org/ticket/46349))

Implements the design [here](https://www.figma.com/proto/WB24a8gm07lJKqLmVr01ZttB/Change-admin-email-v1?node-id=1%3A2&scaling=min-zoom&redirected=1), by boemedia,

## How it works
- A new filter function `wp_confirm_admin_email` hooks into `login_redirect`
  - and returns the normal redirect, unless
  - the email has not been confirmed in `ADMIN_EMAIL_MAX_AGE` time
    - The login process is redirected to `action=confirm_admin_email`
    - Which checks for `manage_options`
      - redirects to normal redirect if user doesn't have the cap. or
      - displays the admin email confirm screen, otherwise
	- The admin email confirm screen, in `wp-login.php`
	  - postpones the confirm, with a remind me later link
	    - which sets an `option` named `admin_email_lifespan`, also
	  - submits a confirm input
	    - which sets an `option` named `admin_email_lifespan`
  - The user end up redirected to the original intented redirect

## How to test

Switch to the branch and update in `default-constants.php` the `ADMIN_EMAIL_MAX_AGE` to a number that makes sense for testing. Make sure your local WP is able to send emails. Login and you should see the new screen.
